app/models/user.rb

```diff
@@ -19,6 +19,10 @@ def superadmin?
     return has_role? :superadmin
   end
 
+  def generate_password
+    ((1..9).to_a+('a'..'z').to_a).shuffle[0,10].join
```

Comment:
It's just generating a random string. I'll add the capital letters too to make it more secure.